### PR TITLE
meson: use template to declare executables (part 6)

### PR DESCRIPTION
**Pull request by yuwata**
Split-out of #28042.

### CODE REVIEW

File: test/meson.build [hunk 1]
```diff
@@ -49,56 +49,188 @@ if install_tests
...
-test_fstab_generator_sh = find_program('test-fstab-generator.sh')
-test_network_generator_conversion_sh = find_program('test-network-generator-conversion.sh')
-test_systemctl_enable_sh = find_program('test-systemctl-enable.sh')
-test_systemd_tmpfiles_py = find_program('test-systemd-tmpfiles.py')
-hwdb_test_sh = find_program('hwdb-test.sh')
+############################################################
+
+if conf.get('ENABLE_HWDB') == 1
+        hwdb_test_sh = find_program('hwdb-test.sh')
+        if want_tests != 'false'
```
Comment (medhefgo) [inline test/meson.build]: These double indentations could be avoided, no?

`if conf.get('ENABLE_HWDB') == 1 and want_tests != 'false'`

File: meson.build [hunk 2]
```diff
@@ -1689,6 +1689,9 @@ if dbus_interfaces_dir == '' or dbus_interfaces_dir == 'yes'
         endif
 endif
 
+dmi_arches = ['x86', 'x86_64', 'aarch64', 'arm', 'ia64', 'loongarch64', 'mips']
+conf.set10('HAVE_DMI', dmi_arches.contains(host_machine.cpu_family()))
```
Comment (medhefgo) [inline meson.build]:
```suggestion
conf.set10('HAVE_DMI', host_machine.cpu_family() in dmi_arches)
```

File: src/test/meson.build [hunk 3]
```diff
@@ -195,194 +193,205 @@ common_test_dependencies = [
...
                 'timeout' : 120,
         },
-        {
+        test_template + {
                 'sources' : files('test-sbat.c'),
-                'condition' : 'ENABLE_BOOTLOADER',
-                'c_args' : '-I@0@'.format(efi_config_h_dir),
+                'conditions' : ['ENABLE_BOOTLOADER'],
+                'c_args' : [
+                        test_cflags,
```
Comment (medhefgo) [inline src/test/meson.build]: This is provided by `test_additional_kwargs`, no?

File: src/test/meson.build [hunk 4]
```diff
@@ -399,17 +408,18 @@ tests += [
...
                 ],
         },
-        {
+        test_template + {
                 'sources' : files('test-time-util.c'),
                 'timeout' : 120,
         },
-        {
+        test_template + {
+                'name' : 'udev-rule-runner',
```
Comment (medhefgo) [inline src/test/meson.build]: Without `test-` prefix it won't get detected as a test anymore, no?

**Comment by medhefgo** [comment]
Looks really nice. Please make sure a `meson test` call reports the same number of tests before and after this PR.

**Comment by yuwata** [comment]
@medhefgo Thank you for the review.

> Looks really nice. Please make sure a `meson test` call reports the same number of tests before and after this PR.

before:
```
Ok:                 1533
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            17  
Timeout:            0   
```
after:
```
Ok:                 1534
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            17  
Timeout:            0   
```

The one difference is now the test-sysv-generator is reintroduced which was mistakenly dropped by 6c713961ab0831fe744a2df9c4e9e258b6ba3105

**Review by medhefgo — APPROVED** [review]

**Pull request merged**